dask/compose.py

```diff
@@ -0,0 +1,105 @@
+import operator
+from functools import partial, wraps
+from itertools import chain
+
+from toolz import merge
+
+from .core import preorder_traversal, istask
+from .optimize import cull
+from .context import _globals
+from . import threaded
+
+
+def get_dasks(task):
+    dsks = chain.from_iterable(t._dasks for t in preorder_traversal(task)
+                               if isinstance(t, Value))
+    return list(dict((id(d), d) for d in dsks).values())
+
+
+def insert_lists(task):
+    if isinstance(task, list):
+        return (list, [insert_lists(i) for i in task])
+    elif istask(task):
+        return (task[0],) + tuple(insert_lists(i) for i in task[1:])
+    else:
+        return task
```

Comment:
We might want to extend this function beyond just lists to include tuples, sets, Iterators, and dicts.  We can probably handle this without having to teach the schedulers about these things by doing things like 

```
if isinstance(task, tuple):
    return (tuple, insert_lists(list(task)))
if isinstance(task, dict):
    return (dict, insert_lists(list(map(list, task.items()))))
if isinstance(task, Iterator):
    return [insert_lists(i) for i in task]  # note that we don't wrap this with list
...
```